File: crates/ecp-cli/tests/fixtures/incremental_parity/ruby/string_utils.rb
module StringUtils
  def self.slugify(text)
    text.downcase.gsub(/[^a-z0-9]+/, '-').gsub(/^-|-$/, '')
  end

  def self.valid_email?(email)
    email.include?('@') && email.split('@').last.include?('.')
  end

  def self.truncate(text, max_len)
    text.length <= max_len ? text : "#{text[0, max_len]}..."
  end

  def self.capitalize_words(text)
    text.split.map(&:capitalize).join(' ')
  end
end

File: crates/ecp-cli/tests/fixtures/incremental_parity/java/StringUtils.java
package com.example;

public class StringUtils {
    public static String slugify(String text) {
        return text.toLowerCase().replaceAll("[^a-z0-9]+", "-").replaceAll("^-|-$", "");
    }

    public static boolean isValidEmail(String email) {
        return email != null && email.contains("@") && email.contains(".");
    }

    public static String truncate(String text, int maxLen) {
        if (text.length() <= maxLen) return text;
        return text.substring(0, maxLen) + "...";
    }

    public static String capitalize(String text) {
        if (text == null || text.isEmpty()) return text;
        return Character.toUpperCase(text.charAt(0)) + text.substring(1);
    }
}
